src/Tools/dotnet-monitor/StreamAccessorService.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Diagnostics.Monitoring.Contracts;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Diagnostics.Monitoring
+{
+    /// <summary>
+    /// Provides direct access to the Response stream for underlying services.
+    /// </summary>
+    internal sealed class StreamAccessorService : IStreamAccessor
+    {
+        private readonly IHttpContextAccessor _contextAccessor;
+        public StreamAccessorService(IHttpContextAccessor contextAccessor)
+        {
+            _contextAccessor = contextAccessor;
+        }
+
+        public Stream OutputStream => _contextAccessor.HttpContext?.Response.Body;
```

Comment:
Does the null coalescing operator apply automatically to all downstream calls as well? If it doesn, my concern is if HttpContext returns null, then _contextAccessor.HttpContext?.Response would return null (due to null coalescing) but then the Body property call will throw an NRE.

Reply:
I just verified that it works properly. (which makes sense since it can short circuit after the first property is null)